<launch>
  <arg name="drone_id" default="0" />
  <arg name="init_x" default="-3"/>
  <arg name="init_y" default="0"/>
  <arg name="init_z" default="0"/>
  <arg name="init_qx" default="1"/>
  <arg name="init_qy" default="0"/>
  <arg name="init_qz" default="0"/>
  <arg name="init_qw" default="1"/>
  <arg name="goal_x" default="3"/>
  <arg name="goal_y" default="0"/>
  <arg name="goal_z" default="1"/>
  <arg name="map_size_x" default="10"/>
  <arg name="map_size_y" default="10"/>
  <arg name="map_size_z" default="6"/>
  <arg name="num_robots" default="3" />

  <arg name="pose_topic" value="mavros/vision_pose/pose"/>
  <arg name="odom_topic" value="mavros/local_position/odom"/>
  <arg name="cloud_topic" value="/drone1/depth/color/points" />
  <arg name="is_pose_subscribed" value="false" />

  <node pkg="plan_manager" name="planner" type="baseline">
    <!-- Topics -->
    <remap from="~pose" to="$(arg pose_topic)" />
    <remap from="~map/odom" to="$(arg odom_topic)" />
    <remap from="~map/cloud" to="$(arg cloud_topic)" />
    <remap from="~map/pose" to="$(arg pose_topic)" />
    <rosparam file="$(find plan_manager)/config/real_cfg.yaml" command="load" />
    <param name="drone_id" value="$(arg drone_id)" />
    <param name="swarm/num_robots" value="$(arg num_robots)" />
    <param name="init_x" value="$(arg init_x)" />
    <param name="init_y" value="$(arg init_y)" />
    <param name="init_z" value="$(arg init_z)" />
    <param name="use_preset_goal" value="true" />
    <param name="goal_x" value="$(arg goal_x)" />
    <param name="goal_y" value="$(arg goal_y)" />
    <param name="goal_z" value="$(arg goal_z)" />
    <param name="vel" value="1" />
    <param name="time_step" value="0.01" />
    <param name="is_pose_sub" value="$(arg is_pose_subscribed)" />
  </node>


  <node pkg="traj_server" name="traj_server" type="bezier_traj_server" output="screen">
    <remap from="~trajectory" to="planner/trajectory" />
    <remap from="/position_cmd" to="controller/pos_cmd" />
    <remap from="~odom" to="$(arg pose_topic)" />
    <param name="init_x" value="$(arg init_x)" />
    <param name="init_y" value="$(arg init_y)" />
    <param name="init_z" value="$(arg init_z)" />
    <param name="init_qx" value="$(arg init_qx)" />
    <param name="init_qy" value="$(arg init_qy)" />
    <param name="init_qz" value="$(arg init_qz)" />
    <param name="init_qw" value="$(arg init_qw)" />
    <param name="tf" value="true" />
    <param name="replan_time_threshold" value="1.2" />
  </node>

</launch>
